### ISSUE TITLE
Avoid CSS values for font-size and line-height that are subject to browsers rounding

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1347]
Just noticed there's a trend in the current Gutenberg CSS to use font-size and line-height values in relative units (noting wrong with this) that result in computed values subject to browsers rounding, for example:
```
.editor-post-visibility__dialog-legend {
	font-weight: 600;
	font-size: 0.9em;
}
```

In this specific case (the post visibility popover "legend") the computed CSS value in Chrome is 11.7px for the font-size and 18.2px for the line-height.

Combined with native fonts, given that every operating system will provide different fonts to WordPress, this exposes to browsers rounding and subtle inconsistencies that will be very hard to fix.

Ideally, values in em should be calculated to compute to a px value without decimals.

Also, 11 or 12 px is a font size WordPress doesn't use, normally. The default font-size is 13px and maybe it's time to increase it a bit too.

**Comment by ntwb** [comment]
@afercia The `line-height` issue will be fixed as part of #1151 

Unitless [line-height](https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#selectors) is already documented in the coding standards, shoud we add for `font-size` to not use `em` or `rem` values (https://developer.mozilla.org/en-US/docs/Web/CSS/font-size#Ems)

p.s. I'll hopefully get to revisit #1151 today/tomorrow to update noted concerns

**Comment by afercia** [comment]
@netweb I'm fine with relative units, they also allow better scaling when users set a different default font size in their operating system. My point is more about rounding. Browser use different rounding algorithms. For example in this case, to have a computed value of `12px` without decimal digits:
Chrome needs 0.923076em
Firefox needs just 0.923em
In other cases I've observed Chrome needs 8 decimals to compute a value without decimals. Combining this with several other factors, sub-pixel rendering, different font faces in different platforms, different font rendering engines, etc. the actual rendering can really be different and make impossible to have perfect alignments across browsers/platforms.

This is something we've observed after the system fonts change in WordPress, see for example https://core.trac.wordpress.org/ticket/36753#comment:84
Probably the only way to limit these differences is making sure any elements always have font-size and line-height values that compute to "full" numbers without decimals.

**Comment by ntwb** [comment]
Right, thanks for the detailed explanation 👍 

> _"Probably the only way to limit these differences is making sure any elements always have font-size and line-height values that compute to "full" numbers without decimals."_

I'll have a think about this, might be able to create a stylelint-plugin to report on this 🤔 

p.s. I know this isn't easily remembered but my username on GitHub is @ntwb

**Comment by jasmussen** [comment]
In that PR above, I "fixed" this issue. That is, I removed the offending 0.9em font size, because I fully agree with the assesment. 

I also noted in that PR that em units can be used for good as well as bad — this ticket definitely found an instance of "bad". But removing all ems isn't a good fix either as it can be a very useful unit for some designs. 

As such if there are other instances, or future instances, where this issue pops up, we should address these on a case by case basis. In addition to that I have made a note to add best practices to the pattern library that I need to build one of these days. Part of that is to not invent new font sizes, because the fewer font sizes, the better.
